app/grandchallenge/cases/models.py

```diff
@@ -271,6 +272,8 @@ def __str__(self):
     @property
     def shape_without_color(self) -> List[int]:
         result = []
+        if self.timepoints is not None:
+            result.append(self.timepoints)
```

Comment:
Should timepoints `t` be returned before depth `z`? I thought that `[x, y, (z), (t)]` would be the logical order but maybe this is done differently in the libraries we use? 

Another thing, is there a reason why we're not including the `resolution_levels` (`u` in pyramid formats)?

Reply:
Ah, I assumed that the shape method on the Cases model should follow numpy ordering, so: `[t, z, y, x]`, because of this [test ](https://github.com/comic/grand-challenge.org/blob/d6cef76cb85affc9ad0016c3b286bfba7cc11697/app/tests/cases_tests/test_background_tasks.py#L163L186) in test_background_tasks.py. However, I just noticed that it is based on the `image5x6x7.mhd` file which is the only mhd file in the test files that has its order in the name and the actual shape order reversed! We might want to have a better look at what order is required for grand-challenge and fix the image5x6x7 image and ordering.

Regarding `resolution_levels` I just do not know, what its purpose is and what it should represent. Is this only for pyramid formats? If so my guess is that these aren't so common for MHD and MHA files. At least there is no default tag for this in the headers.

Reply:
I just spoke with Paul, the ordering is indeed supposed to return `numpy` ordering using the `shape` function, otherwise these things would show the wrong way on grand-challenge right now.  The test-file indeed had inverted shape ordering, so I'll fix that and I'll write a test which asserts that the shape order is the inverse of the get_simplei_itk object retrieved from the model and document the ordering: `[(t), (z), y, x, (c)]`